core/extensions.js

```diff
@@ -135,14 +121,160 @@ Blockly.Extensions.registerMutator = function(name, mixinObj, opt_blockList) {
+ * These functions are 'compose' and 'decompose'.  If a block has one, it must
+ * have both.
+ * @param {!Object} object The object to check.
+ * @param {string} errorPrefix The string to prepend to any error message.
+ * @return {boolean} True if the object has both functions.  False if it has
+ *     neither function.
+ * @throws {Error} if the object has only one of the functions.
+ */
+Blockly.Extensions.checkMutatorDialog = function(object, errorPrefix) {
+  var hasCompose = object.hasOwnProperty('compose');
+  var hasDecompose = object.hasOwnProperty('decompose');
+
+  if (hasCompose && hasDecompose) {
+    if (typeof object['compose'] !== "function") {
+      throw new Error(errorPrefix + 'compose must be a function.');
+    } else if (typeof object['decompose'] !== "function") {
+      throw new Error(errorPrefix + 'decompose must be a function.');
+    }
+    return true;
+  } else if (!hasCompose && !hasDecompose) {
+    return false;
+  } else {
+    throw new Error(errorPrefix +
+        'Must have both or neither of "compose" and "decompose"');
+  }
+};
+
+/**
+ * Check that a block has required mutator properties.  This should be called
+ * after applying a mutation extension.
+ * @param {string} errorPrefix The string to prepend to any error message.
+ * @param {!Blockly.Block} block The block to inspect.
+ */
+Blockly.Extensions.checkBlockHasMutatorProperties = function(errorPrefix,
+    block) {
+  if (!block.hasOwnProperty('domToMutation')) {
+    throw new Error(errorPrefix + 'Applying a mutator didn\'t add "domToMutation"');
+  }
+  if (!block.hasOwnProperty('mutationToDom')) {
+    throw new Error(errorPrefix + 'Applying a mutator didn\'t add "mutationToDom"');
+  }
+
+  // A block with a mutator isn't required to have a mutation dialog, but
+  // it should still have both or neither of compose and decompose.
+  Blockly.Extensions.checkMutatorDialog(block, errorPrefix);
+};
+
+/**
+ * Get a list of values of mutator properties on the given block.
+ * @param {!Blockly.Block} block The block to inspect.
+ * @return {!Array.<Object>} a list with all of the properties, which should be
+ *     functions or undefined, but are not guaranteed to be.
+ */
+Blockly.Extensions.getMutatorProperties = function(block) {
+  var result = [];
+  for (var i = 0; i < Blockly.Extensions.MUTATOR_PROPERTIES_.length; i++) {
+    result.push(block[Blockly.Extensions.MUTATOR_PROPERTIES_[i]]);
+  }
+  return result;
+};
+
+/**
+ * Check that the current mutator properties match a list of old mutator
+ * properties.  This should be called after applying a non-mutator extension,
+ * to verify that the extension didn't change properties it shouldn't.
+ * @param {!Array.<Object>} oldProperties The old values to compare to.
+ * @param {!Blockly.Block} block The block to inspect for new values.
+ * @return {boolean} True if the property lists match.
+ */
+Blockly.Extensions.mutatorPropertiesMatch = function(oldProperties, block) {
+  var match = true;
```

Comment:
Alternatively, you can use an early return false in both cases, instead of a var.